man/fi_getinfo.3.md

```diff
@@ -512,6 +512,15 @@ input flags.  Valid flags include the following.
   the node and/or service parameter must be non-NULL.  This flag is
   often used with passive endpoints.
 
+*FI_PROV_ATTR_ONLY*
+: Indicates that the caller is only querying for what providers are
+  potentially available.  All providers will return exactly one
+  fi_info struct, regardless of whether that provider is usable on the
+  current platform or not.  The returned fi_info struct will contain
```

Comment:
@shefty I think the statement I already have in here covers the situation: "All providers will return exactly one fi_info struct, regardless of whether that provider is usable on the current platform or not."

@sayantansur You made the following comments in a prior commit (that github is now hiding from us):

> Also, I'm not sure what this means:
> 
> "All providers will return exactly one fi_info struct, regardless of whether that provider is usable on the current platform or not."
> 
> Providers might look for drivers loaded etc. in their ini() functions. Failing that, they may want to remove themselves from the available list.
>
> Seems to me if the above were the real requirement, this should be implemented by the framework by simply looking what providers it was compiled with.

This has been updated to be framework-only -- we don't call anything down in the providers.  But the providers must actually be present / have registered in the process.  Otherwise, we can't know what they are (e.g., if some DLs are loaded that libfabric core didn't know about when it was compiled).

Are you good with that?